src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -152,7 +152,10 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             // Ensure the options persisters are loaded since we have to fetch options from the shell
             LoadOptionPersistersAsync(this.ComponentModel, cancellationToken).Forget();
 
+            // Explicitly switch to background so we do the MEF loads off of the UI thread.
+            await TaskScheduler.Default;
             _workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
```

Comment:
nit: Maybe we could add a "GetServiceOnBackgroundThreadAsync<T>" to do this? and then it would just be 

`_workspace = await GetServiceOnBackgroundThreadAsync<VisualStudioWorkspace>(cancellationToken).ConfigureAwait(true)`